packages/ag-charts-enterprise/src/features/annotations/scenes/linearScene.ts

```diff
@@ -66,17 +66,17 @@ export abstract class LinearScene<
         };
     }
 
-    public drag(datum: Datum, target: Coords, context: AnnotationContext) {
+    public drag(datum: Datum, target: Coords, context: AnnotationContext, shiftKey: boolean) {
```

Comment:
The state of whether the shift key is pressed down while clicking/dragging should be handled by the `AnnotationStateMachine`, not the scenes.

Then the `AnnotationStateMachine` could provide a util on the `AnnotationStateContext` like `ctx.isSnapping()` to the other state machines.

I believe this would also allow `point` to remain as a simple value instead of a function. Passing a function into the state transitions is really not ideal and should be avoided. It's creating a lot of mess with unrelated code.

Reply:
Ok I've removed the shiftKey being passed down and now we get the value from keyDown and keyUp events rather than mouse events